subsys/bluetooth/controller/ll_sw/ull_master.c

```diff
@@ -93,6 +93,18 @@ uint8_t ll_create_connection(uint16_t scan_interval, uint16_t scan_window,
 		return BT_HCI_ERR_CMD_DISALLOWED;
 	}
 
+#if defined(CONFIG_BT_CTLR_CHECK_SAME_PEER_CONN)
+	uint8_t own_id_addr_type = (own_addr_type & 0x01);
+	uint8_t *own_id_addr;
+
+	/* Do not connect twice to the same peer */
+	own_id_addr = ll_addr_get(own_id_addr_type, NULL);
+	if (ull_conn_peer_connected(own_id_addr_type, own_id_addr,
+				    peer_addr_type, peer_addr)) {
+		return BT_HCI_ERR_CONN_ALREADY_EXISTS;
+	}
+#endif /* CONFIG_BT_CTLR_CHECK_SAME_PEER_CONN */
+
```

Comment:
@wopu-ot I should resolved the supplied `peer_addr` and store the `peer_id_addr` instead, right?